### PR TITLE
AMP-91849 Turn all arguments into optional to make it easy update script argumetns in future. Add validations for arguments as well.

### PR DESCRIPTION
**Pull request by fzqgriffin**
Turn all arguments into optional to make it easy update script argumetns in future. Add validations for arguments as well.

### CODE REVIEW

File: unload_databricks_data_to_s3.py [hunk 1]
```diff
@@ -76,26 +76,52 @@ def export_meta_data(event_count: int, partition_count: int):
...
+    if args.secret_key_name_for_aws_secret_key is None:
+        raise Exception("Argument --secret_key_name_for_aws_secret_key is None")
+    if args.secret_key_name_for_aws_session_token is None:
+        raise Exception("Argument --secret_key_name_for_aws_session_token is None")
+    if args.s3_region is None:
+        raise Exception("Argument --s3_region is None")
+    if args.s3_path is None:
+        raise Exception("Argument --s3_path is None")
 
 if __name__ == '__main__':
```
Comment (damahua) [inline unload_databricks_data_to_s3.py:99]: is it possible to add a valid command example for reference?

**Review by damahua — APPROVED** [review]
LGTM

File: unload_databricks_data_to_s3.py (hunk 1 above)
Comment (fzqgriffin) [reply unload_databricks_data_to_s3.py:99]: yep, can add that

File: unload_databricks_data_to_s3.py
Comment (shawn-amplitude) [inline unload_databricks_data_to_s3.py:1]: If we only do the None check, we can utilize `required=True` when `parser.add_argument()`.

Comment (fzqgriffin) [reply unload_databricks_data_to_s3.py:1]: good point, let me update this.

**Pull request merged**